### PR TITLE
Fix bug that cause wrong network topology

### PR DESCRIPTION
**Pull request by yilunzhang**
1. If connection is closed when a remote node is in connecting list, the remote node should be removed from connecting list, otherwise the remote node will never be connected again.
2. When comparing chord id, shorter length id should always be less than longer id (effectively add zero padding in front of chord id that is shorter than hashbit).

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [ ] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [ ] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: net/node/node.go [hunk 1]
```diff
@@ -147,6 +146,7 @@ func NewNode() *node {
 	n := node{
 		state: INIT,
 	}
+	n.time = time.Now()
```
Comment (gdmmx) [inline net/node/node.go]: Just a suggestion：
How about below？
```
	n := node{
 		state: INIT,
                time:  time.Now()，
 	}
```

Comment (yilunzhang) [reply net/node/node.go]: It didn't work, because time is a property of node.link but not node.

File: net/node/node.go [hunk 2]
```diff
@@ -658,8 +659,10 @@ func (node *node) ConnectNeighbors() {
 		if !node.IsChordAddrInNeighbors(chordNbr.Id) {
 			chordNbrAddr, err := chordNbr.NodeAddr()
 			if err != nil {
+				log.Error("Get chord neighbor node addr error:", err)
```
Comment (gdmmx) [inline net/node/node.go]: There was logged err in chordNbr.NodeAddr() already.  [NodeAddr()](https://github.com/nknorg/nkn/blob/master/net/chord/vnode.go#L26)
Why log it again at here?

File: net/node/node.go (hunk 1 above)
Comment (gdmmx) [reply net/node/node.go]: My mistake. Sorry about that.

File: net/node/node.go (hunk 2 above)
Comment (yilunzhang) [reply net/node/node.go]: You are right, removing it

**Pull request merged**